### ISSUE TITLE
More contour lines?

### ISSUE THREAD

**Issue by Jadowo** [sh4rkman/SquadCalc#205]

**Comment by Jadowo** [comment]
It would be nice if the circle for the radius also shows the proper max ranges instead of a generic circle.

**Comment by sh4rkman** [comment]
Hi ! Thanks for your interest

Do you mean this kind of concentric circles ?
![image](https://github.com/user-attachments/assets/e1a0d17a-926b-4f61-965a-a98032adfab7)

If yes it would be either be either not taking elevation into account and be imprecise (like the maxrange circle) or taking elevation into account and be **very** CPU intensive, probably crashing the browser. Achieving this kind of circles means reading the heightmap in every direction (360 times at least to draw a decent circle) and that is heavy, specially when dragging a weapon around the map.

@Devil4ngle wrote the code adaptating maxrange circle to real range distance you can observe it live in [this chinese clone](https://squadee.org/). I'm still wondering if i want to do the same because i don't find it _that_ usefull and on a purely cosmetic point of view i find it horrible.

The right approach to this kind of thing is the xyz way. local grid on each mortar arbitrary drawn without taking elevation into account. But it can be hard to read considering squadcalc is way more complex with weapons using degrees (i might find myself drawing line for 10mil and 0.6° ?)

I might think about it but idk where even to start like what leaflet object to use to draw this kind of grid

**Comment by Jadowo** [comment]
Yeah so something like Devil4ngle's max range but for more lines like you show in the picture. But yeah having a full circle is more than likely excessive, doing it like xyz would probably make sense
